sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java

```diff
@@ -1104,13 +1104,17 @@ public void finalize(Iterable<FileResult> writerResults, PipelineOptions options
           tempFiles.add(result.getFilename());
         }
         if (!tempFiles.isEmpty()) {
-          inserter.load(
-              bigQuerySink.jobIdToken,
-              JSON_FACTORY.fromString(bigQuerySink.jsonTable, TableReference.class),
-              tempFiles,
-              JSON_FACTORY.fromString(bigQuerySink.jsonSchema, TableSchema.class),
-              bigQuerySink.writeDisposition,
-              bigQuerySink.createDisposition);
+          try {
+            inserter.load(
+                bigQuerySink.jobIdToken,
+                JSON_FACTORY.fromString(bigQuerySink.jsonTable, TableReference.class),
+                tempFiles,
+                JSON_FACTORY.fromString(bigQuerySink.jsonSchema, TableSchema.class),
+                bigQuerySink.writeDisposition,
+                bigQuerySink.createDisposition);
+          } finally {
+            removeTemporaryFiles(options);
```

Comment:
We should think about this. I would have thought that `tempFiles` is exactly the set of files to remove. If not, what should it be?